### PR TITLE
FLW-3245 swap transaction details subscan added

### PR DESCRIPTION
**Pull request by bnsports**

### CODE REVIEW

File: fearless/Modules/AllDone/AllDoneViewController.swift [hunk 1]
```diff
@@ -77,6 +77,7 @@ extension AllDoneViewController: AllDoneViewInput {
 
     func didReceive(explorer: ChainModel.ExternalApiExplorer?) {
         rootView.subscanButton.isHidden = explorer == nil
```
Comment (Viscount88) [inline fearless/Modules/AllDone/AllDoneViewController.swift:79]: I suggest group these buttons inside custom component with "func updateVisibility(for explorer: ChainModel.ExternalApiExplorer?)". Reuse this component for SwapTransactionDetail module

File: fearless/Modules/SwapTransactionDetail/SwapTransactionDetailRouter.swift [hunk 2]
```diff
@@ -1,3 +1,8 @@
 import Foundation
 
-final class SwapTransactionDetailRouter: SwapTransactionDetailRouterInput {}
+final class SwapTransactionDetailRouter: SwapTransactionDetailRouterInput {
```
Comment (Viscount88) [inline fearless/Modules/SwapTransactionDetail/SwapTransactionDetailRouter.swift:3]: Just use WebPresentable protocol. It's doing same thing.

File: fearless/Modules/SwapTransactionDetail/SwapTransactionDetailViewController.swift [hunk 3]
```diff
@@ -54,6 +66,11 @@ extension SwapTransactionDetailViewController: SwapTransactionDetailViewInput {
     func didReceive(viewModel: SwapTransactionViewModel) {
         rootView.bind(viewModel: viewModel)
     }
+
+    func didReceive(explorer: ChainModel.ExternalApiExplorer?) {
+        rootView.subscanButton.isHidden = explorer == nil
```
Comment (Viscount88) [inline fearless/Modules/SwapTransactionDetail/SwapTransactionDetailViewController.swift:71]: Reuse custom component here.

File: fearless/Modules/SwapTransactionDetail/SwapTransactionDetailViewLayout.swift [hunk 4]
```diff
@@ -176,10 +229,44 @@ final class SwapTransactionDetailViewLayout: UIView {
             makeCommonConstraints(for: view)
         }
 
-        closeButton.snp.makeConstraints { make in
-            make.leading.trailing.equalToSuperview().inset(UIConstants.bigOffset)
+        transactionHashView.snp.makeConstraints { make in
+            make.height.equalTo(Constants.multiViewHeight)
```
Comment (Viscount88) [inline fearless/Modules/SwapTransactionDetail/SwapTransactionDetailViewLayout.swift:233]: I suggest use makeCommonConstraints(for: view) for defining height of each view)

File: fearless/Modules/SwapTransactionDetail/SwapTransactionDetailViewController.swift [hunk 5]
```diff
@@ -54,6 +66,11 @@ extension SwapTransactionDetailViewController: SwapTransactionDetailViewInput {
     func didReceive(viewModel: SwapTransactionViewModel) {
         rootView.bind(viewModel: viewModel)
     }
+
+    func didReceive(explorer: ChainModel.ExternalApiExplorer?) {
```
Comment (DRadmir) [inline fearless/Modules/SwapTransactionDetail/SwapTransactionDetailViewController.swift:70]: I suggest replace ChainModel.ExternalApiExplorer to Bool

**Review by DRadmir — CHANGES_REQUESTED** [review]

**Review by DRadmir — APPROVED** [review]

**Pull request merged**